orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/ecs/EcsServerGroupCreator.groovy

```diff
@@ -86,6 +130,55 @@ class EcsServerGroupCreator implements ServerGroupCreator, DeploymentDetailsAwar
     return [[(ServerGroupCreator.OPERATION): operation]]
   }
 
+  private Supplier<Iterable<Object>> fetchAndParseArtifact(Artifact artifact) {
+    return { ->
+      Response artifactText = oortService.fetchArtifact(artifact)
+      try {
+        if(artifactText != null && artifactText.getBody() != null){
+          return yamlParser.get().loadAll(artifactText.getBody().in());
+        } else{
+          throw new ConfigurationException("Invalid artifact configuration or task definition artifact is null")
+        }
+      } catch (Exception e) {
+        throw new IllegalStateException(e);
+      }
+    }
+  }
+
+  private List<Map<Object, Object>> coerceArtifactToList(Object artifact) {
+    if (artifact instanceof List) {
```

Comment:
when would a task definition artifact contain an instance of a list? wouldn't this indicate an error state, since it needs to be a [RegisterTaskDefinitionRequest](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_RegisterTaskDefinition.html) object?

Reply:
I have changed the type of the evaluated artifact from List to Map, so now it resolves this issue as well.